java-frontend/src/test/files/se/OptionalGetBeforeIsPresentCheck.java

```diff
@@ -81,6 +81,22 @@ private void usingOfNullableWithTest(@Nullable Object o) {
       op.get(); // Compliant - if o is not null, then the optional is necessarily present
     }
   }
+
+  private void usingFilter1(Optional<String> op) {
+    if (op.filter(this::testSomething).isPresent()) {
+      op.get(); // Compliant - filter should return the same optional if test pass
+    }
+    op.get(); // Noncompliant
+  }
+
+  private void usingFilter2(Optional<String> op) {
+    if (!op.filter(this::testSomething).isPresent()) {
+      return;
+    }
+    op.get(); // Compliant - FN
+  }
+
+  abstract boolean testSomething(String s);
```

Comment:
you can use `String::isEmpty` for the same effect and no need to make class abstract